lib/puppet/provider/foreman_smartproxy/rest_v3.rb

```diff
@@ -4,7 +4,7 @@
   def proxy
     @proxy ||= begin
       r = request(:get, 'api/v2/smart_proxies', :search => %{name="#{resource[:name]}"})
-      raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved: #{error_message(r)}") unless success?(r)
+      raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved from Foreman: #{error_message(r)}") unless success?(r)
```

Comment:
Should we suggest to consult the Foreman logs as well?

https://github.com/theforeman/puppet-foreman/blob/746229243abb98995981f90d1a777e096d201108/lib/puppet/provider/foreman_smartproxy/rest_v3.rb#L70 does this